backend/app/api/api_v1/endpoints/reports.py

```
@@ -727,10 +811,39 @@ async def get_report_by_id(
         pct=str(policy_val.get("pct", "100")),
     )
 
+    raw_records = list(report.get("records", []))
+    source_rows = [_source_row_from_report_record(rec) for rec in raw_records]
+    provider = get_default_provider(db)
+    ips = [str(row.get("source_ip") or "unknown") for row in source_rows]
+    hostnames = await asyncio.gather(*[_safe_ptr_lookup(provider, ip) for ip in ips])
+    sender_by_ip = {
+        ip: identify_sender(ip, row, hostname=hostname, domain=report.get("domain", ""))
+        for ip, row, hostname in zip(ips, source_rows, hostnames)
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)